cmd/kube-apiserver/app/testing/testserver.go

```diff
@@ -37,6 +37,12 @@ import (
 // TearDownFunc is to be called to tear down a test server.
 type TearDownFunc func()
 
+// TestServerInstanceOptions Instance options the TestServer
+type TestServerInstanceOptions struct {
+	// Storage cleanup is automated if enabled
+	AutoCleanupStorage bool
```

Comment:
nit: make this default to the old behavior. e.g. `DisableStorageCleanup`

That makes `&TestServerInstanceOptions{}` and `NewDefaultTestServerOptions` return the same values which is kinda what you expect anyway.